packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -127,8 +132,13 @@ export class PropertiesManager {
 		return newProps;
 	}
 
-	public hasPendingProperties() {
-		return Object.keys(this.pendingKeyUpdateCount!).length > 0;
+	public hasPendingProperties(props: PropertySet) {
```

Comment:
better to clarify whether this is an 'all' or 'any':

```suggestion
	/**
	 * @returns whether all valid (i.e. defined) entries of the property bag are pending
	 */
	public hasPendingProperties(props: PropertySet) {
```